explorer/interpreter/stack_space.h

```diff
@@ -0,0 +1,57 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_EXPLORER_INTERPRETER_STACK_SPACE_H_
+#define CARBON_EXPLORER_INTERPRETER_STACK_SPACE_H_
+
+#include <functional>
+#include <optional>
+
+namespace Carbon {
+
+namespace Internal {
+
+// Returns true if a new thread should be started for more stack space.
+auto IsStackSpaceNearlyExhausted() -> bool;
+
+// Starts a thread to run the function.
+auto RunWithStackSpaceHelper(std::function<void()> fn) -> void;
+
+}  // namespace Internal
+
+// Starts a new thread with initialized stack space. This should be called
+// before RunWithStackSpace so that execution is occurring with known stack
+// space.
```

Comment:
This makes it sound like this function doesn't actually run `fn`, it just somehow prepares for the `RunWithStackSpace` call that actually will run `fn`. But judging from the function body, this actually does run `fn`. But if that's the case, the name and comment seem misleading, and I don't understand how to choose between this function and `RunWithStackSpace`

Reply:
I've tried to adjust the comments to make this clearer: Init is used first because it's initialization, and mechanically it's necessary in order to know how much stack space remains.